tests/ert_tests/experiment_server/test_server.py

```diff
@@ -0,0 +1,79 @@
+import sys
+import ert.experiment_server
+import pytest
+
+try:
+    from unittest.mock import AsyncMock
+except ImportError:
+    from mock import AsyncMock
+from ert.experiment_server._experiment_protocol import Experiment
+from typing import AsyncContextManager, Awaitable, Callable
+
+from cloudevents.http import CloudEvent, to_json
+from websockets.client import WebSocketClientProtocol
+
+# All test coroutines will be treated as marked.
+pytestmark = pytest.mark.asyncio
+minversion = pytest.mark.skipif(
+    sys.version_info < (3, 7), reason="requires python3.7 or higher"
+)
+
+
+@minversion
+async def test_receiving_event_from_cluster(
+    experiment_server_ctx: AsyncContextManager[ert.experiment_server.ExperimentServer],
+    dispatcher_factory: AsyncContextManager[
+        Callable[
+            [ert.experiment_server.ExperimentServer], Awaitable[WebSocketClientProtocol]
+        ]
+    ],
+):
+    async with experiment_server_ctx as experiment_server:
+        experiment = AsyncMock(Experiment)
+        experiment_server.add_experiment(experiment)
+
+        async with dispatcher_factory as make_dispatcher:
+            dispatcher = await make_dispatcher(experiment_server)
+            assert experiment_server._dispatchers_connected.qsize() == 1
+
+            event = CloudEvent(
+                {
+                    "type": "test.event",
+                    "source": "test_receiving_event_from_cluster",
+                }
+            )
+            await dispatcher.send(to_json(event))
+
+    experiment.dispatch.assert_awaited_once_with(event, 0)
+
+
+@minversion
+async def test_successful_run(
+    experiment_server_ctx: AsyncContextManager[ert.experiment_server.ExperimentServer],
+    capsys,
+):
+    async with experiment_server_ctx as experiment_server:
+        experiment = AsyncMock(Experiment)
+        experiment.successful_realizations.return_value = 5
+        id_ = experiment_server.add_experiment(experiment)
+        await experiment_server.run_experiment(id_)
+
```

Comment:
Shall we check also whether for instance `EVTYPE_EXPERIMENT_SUCCEEDED` was produced? Not for the sake of test but just to see what is in the workings under the hood :)

Reply:
On this abstraction level (a experiment server), that event is not communicated to clients, only as communication between the experiment and the experiment server.

The production of that event is definitely something we should assert when testing the different experiments, though.